x/attribute/wasm/encode.go

```diff
@@ -91,6 +125,34 @@ func (params *DeleteAttributeParams) Encode(contract sdk.AccAddress) ([]sdk.Msg,
 	return []sdk.Msg{msg}, nil
 }
 
+// Encode creates a MsgDeleteDistinctAttribute.
+// The contract must be the owner of the name of the attribute being deleted.
```

Comment:
Should we be doing this check? Is this the correct place for that check?

Reply:
the validation of the owner will actually happen when the message is processed. this function just creates the message to be emitted from the smart contract and only validates that the address is an actual usable address (`types.ValidateAttributeAddress`)

Reply:
The `AddAttributeParams.Encode` function has this comment too. It's not actually something specifically enforced in this wasm stuff though, it's actually enforced in the attribute module.

Reply:
updated comment for clarification